src/reducer.js

```diff
@@ -0,0 +1,45 @@
+const initialState = {
+  regions: [],
+  categories: [],
+  selectedCategory: '',
+  selectedRegion: '',
+  restaurants: [],
+};
+
+const initialAction = {
+  type: 'initialAction',
+};
+
+const reducers = {
+  setRegions: (state, { regions }) => ({
+    ...state,
+    regions,
+  }),
+  setCategories: (state, { categories }) => ({
+    ...state,
+    categories,
+  }),
+
+  setRestaurants: (state, { restaurants }) => ({
+    ...state,
+    restaurants,
+  }),
+  selectCategory: (state, payload) => ({
```

Comment:
```suggestion
  selectCategory: (state, {id}) => ({
```
다른 메서드 처럼 구조 분해 할당으로 처리할 수 있을 것 같네요~